src/appengine/handlers/cron/project_setup.py

```diff
@@ -740,115 +554,350 @@ def cleanup_pubsub_topics(project_names):
+
+    return (service_account, backup_bucket_name, corpus_bucket_name,
+            logs_bucket_name, quarantine_bucket_name)
+
+  def _get_build_bucket_path(self, project_name, engine, memory_tool,
+                             architecture):
+    """Returns the build bucket path for the |project|, |engine|, |memory_tool|,
+    and |architecture|."""
+    return self._build_bucket_path_template.format(
+        bucket=self._get_build_bucket(engine, architecture),
+        project=project_name,
+        sanitizer=memory_tool)
+
+  def _sync_job(self, project, info, corpus_bucket_name, quarantine_bucket_name,
+                logs_bucket_name, backup_bucket_name):
+    """Sync the config with ClusterFuzz."""
+    # Create/update ClusterFuzz jobs.
+    for template in get_jobs_for_project(project, info):
+      if template.engine == 'none':
+        # Engine-less jobs are not automatically managed.
+        continue
+
+      fuzzer_entity = self._fuzzer_entities.get(template.engine)
+      if not fuzzer_entity:
+        raise ProjectSetupError('Invalid fuzzing engine.')
+
+      job_name = template.job_name(project)
+      job = data_types.Job.query(data_types.Job.name == job_name).get()
+      if not job:
+        job = data_types.Job()
+
+      if job_name not in fuzzer_entity.jobs and not info.get('disabled', False):
+        # Enable new job.
+        fuzzer_entity.jobs.append(job_name)
+
+      job.name = job_name
+      if self._segregate_projects:
+        job.platform = untrusted.platform_name(project, 'linux')
+      else:
+        # TODO(ochang): Support other platforms?
+        job.platform = 'LINUX'
+
+      job.templates = template.cf_job_templates
+
+      revision_vars_url = self._revision_url_template.format(
+          project=project,
+          bucket=self._get_build_bucket(template.engine, template.architecture),
+          sanitizer=template.memory_tool)
+
+      job.environment_string = JOB_TEMPLATE.format(
+          build_bucket_path=self._get_build_bucket_path(
+              project, template.engine, template.memory_tool,
+              template.architecture),
+          engine=template.engine,
+          project=project,
+          revision_vars_url=revision_vars_url)
+
+      if logs_bucket_name:
+        job.environment_string += 'FUZZ_LOGS_BUCKET = {logs_bucket}\n'.format(
+            logs_bucket=logs_bucket_name)
+
+      if corpus_bucket_name:
+        job.environment_string += 'CORPUS_BUCKET = {corpus_bucket}\n'.format(
+            corpus_bucket=corpus_bucket_name)
+
+      if quarantine_bucket_name:
+        job.environment_string += (
+            'QUARANTINE_BUCKET = {quarantine_bucket}\n'.format(
+                quarantine_bucket=quarantine_bucket_name))
+
+      if backup_bucket_name:
+        job.environment_string += 'BACKUP_BUCKET = {backup_bucket}\n'.format(
+            backup_bucket=backup_bucket_name)
+
+      if self._add_additional_labels:
```

Comment:
_add_additional_labels seems like a weird name, maybe maybe _add_info_labels / _add_metadata_labels or something ?

Reply:
renamed to add_info_labels.